compiler/noirc_evaluator/src/acir/call/intrinsics/slice_ops.rs

```diff
@@ -484,6 +609,10 @@ impl Context<'_> {
             "ICE: The read flattened slice should match the computed size"
         );
 
+        let item_size = slice_typ.element_size();
+        let item_size = self.acir_context.add_constant(item_size);
+        let remove_index = self.acir_context.mul_var(remove_index, item_size)?;
```

Comment:
Is this needed? Was there a bug related to this? Same for the other intrinsics where this was changed.